### PR TITLE
Feature 29 vacation spec design

### PR DESCRIPTION
**Pull request by naStipe**
Implemented vacation request  specs design according to time-bank-API-spec

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -1198,6 +1540,119 @@ components:
...
+          description: Vacation days used for requested vacation
+        type:
+          $ref: "#/components/schemas/VacationType"
+        message:
+          type: string
+          description: Message attached to request
+        createdBy:
+          type: string
+          format: uuid
+          description: ID of person who created status in UUID
```
Comment (dalju005-xamk) [inline swagger.yaml:1603]: Person's Keycloak ID ?

File: swagger.yaml [hunk 2]
```diff
@@ -1198,6 +1540,119 @@ components:
           type: boolean
           description: Boolean value to indicate if option is correct or incorrect
 
+    VacationType:
+      type: string
+      description: Enum for vacation type
+      enum:
+        - VACATION
```
Comment (dalju005-xamk) [inline swagger.yaml:1547]: These should be inline with company policies and collective labor agreements.

File: swagger.yaml [hunk 3]
```diff
@@ -750,7 +750,349 @@ paths:
...
+      responses:
+        "204":
+          description: Empty response indicating successful removal.
+        "default":
+          description: Invalid request was sent to the server
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+  "/vacationRequests/{id}/statuses":
```
Comment (dalju005-xamk) [inline swagger.yaml:920]: Not sure are these vacationRequest status thing so needed. Could these be handled some other way ?

**Review by dalju005-xamk — APPROVED** [review]
Nice work dudes. Not sure if all this is needed, but that is something to find out in future.

File: swagger.yaml (hunk 1 above)
Comment (Hanhz2707) [reply swagger.yaml:1603]: yeah, createdBy: / type: string / format: uuid / description: Person keycloak ID in UUID

**Review by AfoninOleksandr — APPROVED** [review]
Looks good

File: swagger.yaml [hunk 4]
```diff
@@ -750,7 +750,349 @@ paths:
...
+            type: boolean
+      responses:
+        '200':
+          description: List of vacation requests.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/VacationRequest'
```
Comment (Jdallos) [inline swagger.yaml]: Lets update all usages of single quotes with double quotes.

File: swagger.yaml (hunk 3 above)
Comment (Jdallos) [reply swagger.yaml:920]: In the original design it was decided that the vacation status should be a separate entity, this allows us to record multiple vacation "responses" e.g. approval or decline along with accompanying messages/ notes.

Feel free to have a think if you want to change this and we could have further discussion. At least IMO vacationRequestStatuses are fine for now as the UI is already build around them being a separate entity.

File: swagger.yaml (hunk 2 above)
Comment (Jdallos) [reply swagger.yaml:1547]: Yes these might need to be updated- I'd suggest sending Taneli or Eetu a message about this current list and asking them what other options are used e.g. in Severa.

File: swagger.yaml [hunk 5]
```diff
@@ -1198,6 +1540,119 @@ components:
...
+      required:
+        - personId
+        - days
+        - startDate
+        - endDate
+        - type
+        - message
+        - createdAt
+        - updatedAt
+        - draft
```
Comment (Jdallos) [inline swagger.yaml]: Maybe this does not need to be a required parameter?

File: swagger.yaml (hunk 1 above)
Comment (Jdallos) [reply swagger.yaml:1603]: This id will be coming from the users schema. The user id is the same as keycloak Id.

**Review by Jdallos — APPROVED** [review]
Approved with some suggestions

File: swagger.yaml (hunk 3 above)
Comment (naStipe) [reply swagger.yaml:920]: Then I guess we will keep them for now

File: swagger.yaml (hunk 4 above)
Comment (Hanhz2707) [reply swagger.yaml]: updated

File: swagger.yaml (hunk 5 above)
Comment (Hanhz2707) [reply swagger.yaml]: deleted

**Pull request merged**